### PR TITLE
Ingest modular

### PR DESCRIPTION
**Pull request by jwildfire**
## Overview
This PR extends the `ingest` branch refactor by splitting all workflows modules that correspond with specific data base outputs.  This resulted in a major refactor of the `inst/workflows` folder and several tweaks to `RunWorkflow(s)` and other associated utility functions. 

The biggest changes overall are: 
-`RunWorkflow()` now returns only the result of it's last `Step` by default
-`RunWorkflows` is now run in sequence based on the `meta$priority` parameter in each workflow
- The list returned by `RunWorkflows` is now named based on a new `strResultNames` parameter that pulls values from `workflow$meta` The default is  `c("Type","ID")` which is parsed to `{meta$Type}_{meta$ID}`

Examples 1, 2 and 3 in `inst/examples/3_ReportingWorkflows.R` have been updated. 

## Test Notes/Sample Code

See examples in `inst/examples/3_ReportingWorkflows.R

### CODE REVIEW

File: inst/workflow/4_modules/report_kri_site.yaml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 meta:
-  File: report_kri_site.yaml
+  Type: Report 
+  ID: report_kri_site
```
Comment (zdz2101) [inline inst/workflow/4_modules/report_kri_site.yaml:3]:
```suggestion
  ID: report_kri_site
```
```suggestion
  ID: report_kri_site
  Output: html
```
To match above `report_kri_country.yaml`

File: inst/examples/3_ReportingWorkflow.R [hunk 2]
```diff
@@ -13,60 +13,85 @@ lData <- list(
...
 )
-ingest_wf <- MakeWorkflowList(strNames = "1_ingest")
-raw <- RunWorkflows(ingest_wf, lData)
 
-# Step 2 - Create Mapped Data - filter, aggregate and join raw data to create mapped data layer
-map_wf <- MakeWorkflowList(strNames = "2_map")
-mapped <- RunWorkflows(map_wf, raw$lRaw)
+# Step 0 - Data Ingestion - standardize tables/columns names 
+lRaw <- list(
+    Raw_SUBJ = Source_SUBJ,
```
Comment (zdz2101) [inline inst/examples/3_ReportingWorkflow.R:21]:
```suggestion
    Raw_SUBJ = lSource$Source_SUBJ,
```

Find and replace "Source_" with "lSource$Source_" for lines 21-43 ?

**Review by zdz2101 — COMMENTED** [review]
what I got so far, I'll take another look tomorrow

File: R/util-MakeMetric.R [hunk 3]
```diff
@@ -22,6 +22,7 @@ MakeMetric <- function(lWorkflows) {
     purrr::map(function(wf) {
       return(tibble::as_tibble(wf$meta))
     }) %>%
-    purrr::list_rbind()
+    purrr::list_rbind() %>%
```
[the remark this answers is not in the document]
Comment (lauramaxwell) [reply R/util-MakeMetric.R:25]: Yes, but we can merge this into `ingest` and then deal with all of the documentation issues prior to merging into `dev`

**Review by lauramaxwell — APPROVED** [review]

**Pull request merged**